snmplib/transports/snmpTLSBaseDomain.c

```diff
@@ -245,88 +245,39 @@ netsnmp_tlsbase_verify_server_cert(SSL *ssl, _netsnmpTLSBaseData *tlsdata) {
-
-            /* see if the requested hostname has a wildcard prefix */
-            if (strncmp(tlsdata->their_hostname, "*.", 2) == 0) {
-                is_wildcarded = 1;
-                compare_to = tlsdata->their_hostname + 2;
-            } else {
-                compare_to = tlsdata->their_hostname;
+            size_t their_hostname_len = strlen(tlsdata->their_hostname);
+            their_hostname = tlsdata->their_hostname;
+            
+            /* RFC 6353: convert their_hostname to lowercase */
+            char *lower_hostname = calloc(their_hostname_len + 1, sizeof(char));
+            if (NULL == lower_hostname) {
+                LOGANDDIE("Failed to allocate memory to convert hostname to lowercase");
             }
-
-            /* if the hostname we were expecting to talk to matches
-               the cert, then we can accept this connection. */
-
-            /* check against the DNS subjectAltName */
-            onames = (GENERAL_NAMES *)X509_get_ext_d2i(remote_cert,
-                                                       NID_subject_alt_name,
-                                                       NULL, NULL );
-            if (NULL != onames) {
-                count = sk_GENERAL_NAME_num(onames);
-
-                for (i=0 ; i <count; ++i)  {
-                    oname = sk_GENERAL_NAME_value(onames, i);
-                    if (GEN_DNS == oname->type) {
-
-                        /* get the value */
-                        ASN1_STRING_to_UTF8((unsigned char**)&check_name,
-                                            oname->d.ia5);
-
-                        /* convert to lowercase for comparisons */
-                        for (j = 0 ;
-                             *check_name && j < sizeof(buf)-1;
-                             ++check_name, ++j ) {
-                            if (isascii(*check_name))
-                                buf[j] = tolower(0xFF & *check_name);
-                        }
-                        if (j < sizeof(buf))
-                            buf[j] = '\0';
-                        check_name = buf;
-                        
-                        if (is_wildcarded) {
-                            /* we *only* allow passing till the first '.' */
-                            /* ie *.example.com can't match a.b.example.com */
-                            check_name = strchr(check_name, '.') + 1;
-                        }
-
-                        DEBUGMSGTL(("tls_x509:verify", "checking subjectAltname of dns:%s\n", check_name));
-                        if (strcmp(compare_to, check_name) == 0) {
-
-                            DEBUGMSGTL(("tls_x509:verify", "Successful match on a subjectAltname of dns:%s\n", check_name));
-                            return SNMPERR_SUCCESS;
-                        }
-                    }
-                }
+            for (i = 0; i < their_hostname_len; ++i) {
+                lower_hostname[i] = tolower((unsigned char) their_hostname[i]);
```

Comment:
The tolower() argument must be a letter. Please add an isalpha() check.

Reply:
As far as I have understood the function tolower() .... isaphla() is done internal. 
https://www.cplusplus.com/reference/cctype/tolower/

Anyway I will add this additional check. It won't hurt :)